### PR TITLE
Upgrade cabal file version

### PR DESCRIPTION
**Pull request by newhoggy**
Testing to see what happens in CI

### CODE REVIEW

File: cardano-api/cardano-api.cabal [hunk 1]
```diff
@@ -1,3 +1,5 @@
+cabal-version: 2.4
```
Comment (erikd) [inline cardano-api/cardano-api.cabal:1]: Probably should be `3.0` because we are using `3.0` features (ie the `cabal.project` file).

Comment (newhoggy) [reply cardano-api/cardano-api.cabal:1]: I tried `3.0` but it failed the build.  For some reason `2.4` is as high as we can go.

Comment (erikd) [reply cardano-api/cardano-api.cabal:1]: Ok, I would be fine leaving it like that for now. The required nix fix is probably large and hairy.

**Comment by newhoggy** [comment]
Thanks!

**Comment by newhoggy** [comment]
bors merge

**Comment by mrBliss** [comment]
I was looking into this again in #1670 and discovered the following:

In `cabal-install-3.2.0.0`, the `cabal-version` field [is parsed as follows](https://github.com/haskell/cabal/blob/329ebb1ad4c132e26b99e128450f9dafa1144133/Cabal/Distribution/PackageDescription/Parsec.hs#L106-L110):

```haskell
[..]
    -- if we get too new version, fail right away
    case ver of
        Just v | v > mkVersion [3,0] -> parseFailure zeroPos
            "Unsupported cabal-version. See https://github.com/haskell/cabal/issues/4899."
        _ -> pure ()
[..]
```

So even version 3.2.0.0 explicitly does not support a version >= 3.0 :roll_eyes:. Although, I discovered that this version number corresponds to the "cabal spec version", which, I suppose, not necessarily has to match the "Cabal/cabal-install version".

This was fixed in 3.4.0.0:

```haskell
        -- if we get too new version, fail right away
        Just v -> case cabalSpecFromVersionDigits (versionNumbers v) of
            Just csv -> return (Just csv)
            Nothing  -> parseFatalFailure zeroPos $
                "Unsupported cabal-version " ++ prettyShow v ++ ". See https://github.com/haskell/cabal/issues/4899."
```

I suppose we'll have to wait until 3.4.0.0 is officially released and wide-spread before we can increase this version number.

**Pull request merged**